documentation/modules/ROOT/partials/modules/all-connectors/custom-snapshotter-spi.adoc

```diff
@@ -0,0 +1,149 @@
+For more advanced uses, you can fine-tune control the snapshot by implementing one of the following interfaces:
+
+`io.debezium.snapshot.spi.Snapshotter`:: Controls whether the connector takes a snapshot.
+`io.debezium.snapshot.spi.SnapshotQuery`:: Controls how data is queried during a snapshot.
+`io.debezium.snapshot.spi.SnapshotLock`:: Controls whether the connector locks tables when taking a snapshot.
+
+
+[source,java,indent=0,subs="+attributes"]
+.io.debezium.snapshot.spi.Snapshotter interface. All built-in snapshot modes implement this interface.
+----
+/**
+ * {@link Snapshotter} is used to determine the following details about the snapshot process:
+ * <p>
+ * - Whether a snapshot occurs. <br>
+ * - Whether streaming continues during the snapshot. <br>
+ * - Whether the snapshot includes schema (if supported). <br>
+ * - Whether to snapshot data or schema following an error.
+ * <p>
+ * Although Debezium provides many default snapshot modes,
+ * to provide more advanced functionality, such as partial snapshots,
+ * you can customize implementation of the interface.
+ * For more information, see the documentation.
+ *
+ *
+ * @author Mario Fiore Vitale
```

Comment:
I think that I missed this before, but is there any reason why you wouldn't want your name in the productized documentation?